File: solutions/hashmaps/205.isomorphic-strings.py
#
# @lc app=leetcode id=205 lang=python
#
# [205] Isomorphic Strings
#

# @lc code=start
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        mydict = {}
        for i in range(len(s)):
            if s[i] not in mydict:
                mydict[s[i]] = t[i]
            elif mydict[s[i]] != t[i]:
                return False
        if len(set(s)) != len(set(t)):
            return False
        return True
        
# @lc code=end

File: solutions_java/hashmaps/205.isomorphic-strings.java
package solutions_java.hashmaps;
/*
 * @lc app=leetcode id=205 lang=java
 *
 * [205] Isomorphic Strings
 */

// @lc code=start
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> smap = new HashMap<>();
        Map<Character, Character> tmap = new HashMap<>();

        for (int i=0; i<s.length(); i++) {
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);

            if (smap.containsKey(sChar)) {
                if (smap.get(sChar) != tChar) {
                    return false;
                }
            } else {
                smap.put(sChar, tChar);
            }

            if (tmap.containsKey(tChar)) {
                if (tmap.get(tChar) != sChar) {
                    return false;
                }
            } else {
                tmap.put(tChar, sChar);
            }
          
        }
        return true;  
    }
}
// @lc code=end

